ballerina/Module.md

```diff
@@ -156,6 +156,43 @@ string|int? generatedKey = result.lastInsertId;
 These samples show how to demonstrate the different usages of the `query` operation to query the
 database table and obtain the results. 
 
+The `ParameterizedQuery` is used to construct the dynamic query to be executed by the client. You can create a simple query like below.
+
+```ballerina
+int id = 10;
+int age = 12;
+ParameterizedQuery query = `SELECT * FROM students WHERE 
+                            id < ${id} AND age > ${age}`;
+```
+
+The `queryConcat()` makes it easier to create a dynamic complex query by concatenating sub-dynamic queries.
+The sample below shows how to concatenate queries.
+
+```ballerina
+int intType = 2147483647;
+int bigIntType = 9223372036854774807;
+int smallIntType = 32767;
+ParameterizedQuery query = `INSERT INTO NumericTypes (int_type,
+                            bigint_type, smallint_type)`;
+ParameterizedQuery query1 = ` VALUES(${intType}, ${bigIntType}, 
+                            ${smallIntType})`;
+ParameterizedQuery sqlQuery = queryConcat(query, query1);
+```
+
+The `arrayFlattenQuery()` util function accepts the array value and returns a parameterized query.
+You can construct the complex dynamic query by using both functions like below.
```

Comment:
Shall we follow the same approach as in slack?. talk abt IN query first and intro this as a helper function to flatten the array